### PR TITLE
mp-b2c-voucher-endpoints-tests

### PR DESCRIPTION
**Pull request by shubham-khandelwal2-metacube**
## PR Description

https://spryker.atlassian.net/browse/CC-16468

### CODE REVIEW

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/negative.robot [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+Add_voucher_code_to_guest_cart_with_invalid_cart_id
+    [Setup]    I set Headers:    X-Anonymous-Customer-Unique-Id=${x_anonymous_prefix}${random}
+    When I send a POST request:    /guest-carts/fake_guest_cart_id/vouchers    {"data": {"type": "vouchers","attributes": {"code": "fake_discount_voucher_code"}}}
+    Then Response status code should be:    404
+    And Response reason should be:    Not Found
+    And Response should return error code:    101
+    And Response should return error message:    Cart with given uuid not found.
+
+Add_voucher_code_from_another_discount_to_cart
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/discount_endpoints/vouchers/negative.robot:130]: what does "from another discount" mean here and below?

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    And Response body parameter should be:    [data][type]    guest-carts
+    And Response body parameter should be:    [data][id]    ${guest_cart_id}
+    And Response body parameter should be:    [data][attributes][priceMode]    ${gross_mode}
+    And Response body parameter should be:    [data][attributes][currency]    ${currency_code_eur}
+    And Response body parameter should be:    [data][attributes][store]    ${store_de}
+    #relationships
+    And Response body parameter should be:    [data][relationships][vouchers][data][0][type]    vouchers
+    And Response body parameter should be:    [data][relationships][vouchers][data][0][id]    ${discount_voucher_code}
+    #included
+    And Response body parameter should be:    [included][0][type]    vouchers
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot:120]: please check includes using standard methods (and not by hardcoding index)
And Response should contain the array of a certain size:    [data][relationships][vouchers][data]    1
And Response include should contain certain entity type:    vouchers
And Response include element has self link:    vouchers

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    And Response body parameter should be:    [data][type]    carts
+    And Response body parameter should be:    [data][id]    ${cart_id}
+    And Response body parameter should be:    [data][attributes][priceMode]    ${gross_mode}
+    And Response body parameter should be:    [data][attributes][currency]    ${currency_code_eur}
+    And Response body parameter should be:    [data][attributes][store]    ${store_de}
+    #relationships
+    And Response body parameter should be:    [data][relationships][vouchers][data][0][type]    vouchers
+    And Response body parameter should be:    [data][relationships][vouchers][data][0][id]    ${discount_voucher_code}
+    #included
+    And Response body parameter should be:    [included][0][type]    vouchers
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot:93]: please check includes using standard methods (and not by hardcoding index)
And Response should contain the array of a certain size:    [data][relationships][vouchers][data]    1
And Response include should contain certain entity type:    vouchers
And Response include element has self link:    vouchers

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+#DELETE requests
+Delete_voucher_code_from_cart
+     [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    Find or create customer cart
+    ...    AND    Cleanup all items in the cart:    ${cart_id}
+    ...    AND    I send a POST request:    /carts/${cart_id}/items    {"data": {"type": "items","attributes": {"sku": "${discount_concrete_product_sku_with_voucher_code}","quantity": 1}}}
+    ...    AND    I send a POST request:    /carts/${cart_id}/vouchers    {"data": {"type": "vouchers","attributes": {"code": "${discount_voucher_code}"}}}
+    When I send a DELETE request:    /carts/${cart_id}/vouchers/${discount_voucher_code}
+    Then Response status code should be:    204
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot:142]: please also check that cart does not have vouchers after the deletion (discount in totals is 0)

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    And Response reason should be:    No Content
+
+Delete_voucher_code_from_guest_user_cart
+    [Setup]    Run Keywords    Create a guest cart:    ${x_anonymous_prefix}${random}    ${discount_concrete_product_sku_with_voucher_code}    1
+    ...   AND    I send a POST request:    /guest-carts/${guest_cart_id}/vouchers    {"data": {"type": "vouchers","attributes": {"code": "${discount_voucher_code}"}}}
+    When I send a DELETE request:    /guest-carts/${guest_cart_id}/vouchers/${discount_voucher_code}
+    Then Response status code should be:    204
+    And Response reason should be:    No Content
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot:150]: please also check that cart does not have vouchers after the deletion (discount in totals is 0)

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/negative.robot (hunk 1 above)
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/discount_endpoints/vouchers/negative.robot:130]: We have 2 voucher discounts as demo data, let’s call them `voucher_discount1 `and `voucher_discount2`
For each discount we have different voucher codes
So, if we check `voucher_discount1` we can set voucher code from the `voucher_discount2`

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot (hunk 4 above)
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot:142]: DONE

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/negative.robot (hunk 1 above)
Comment (OKrivtsova) [reply tests/api/mp_b2c/glue/discount_endpoints/vouchers/negative.robot:130]: I do not see the point of this test - not is a negative test, you do not add any voucher to cart anyway - not for voucher 1 or voucher2. the same goes for Add_voucher_code_from_another_discount_to_guest_user_cart and all other tests with "another"
in all these tests you add "code": "discount_voucher_code" which is not actually a voucher code. these tests do not seem to test anything new

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot (hunk 5 above)
Comment (OKrivtsova) [reply tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot:157]: here again it should not be in teardown

Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot:157]: FIXED

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot (hunk 4 above)
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot:147]: FIXED

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/negative.robot (hunk 1 above)
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/discount_endpoints/vouchers/negative.robot:130]: removed as not required

File: tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot (hunk 3 above)
Comment (OKrivtsova) [reply tests/api/mp_b2c/glue/discount_endpoints/vouchers/positive.robot]: not done. this is nit what needs to be there:
`
    #included
    And Response include should contain certain entity type:    vouchers
    And Response body parameter should be:    [included][0][id]    ${discount_voucher_code}
    And Response body parameter should be greater than:    [included][0][attributes][amount]    0
    And Response body parameter should be:    [included][0][attributes][code]    ${discount_voucher_code}
    And Response body parameter should be:    [included][0][attributes][discountType]    voucher
    And Response body parameter should be:    [included][0][attributes][displayName]    ${discount_3_name}
    And Response body parameter should be:    [included][0][attributes][isExclusive]    False
    And Response body parameter should not be EMPTY:    [included][0][attributes][expirationDateTime]
    And Response body parameter should be:    [included][0][attributes][discountPromotionAbstractSku]    None
    And Response body parameter should be:    [included][0][attributes][discountPromotionQuantity]    None
    And Response include element has self link:    vouchers
`

**Review by OKrivtsova — APPROVED** [review]

**Review by OKrivtsova — COMMENTED** [review]
please merge master into your branch and fix coflicts

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**